oak-store-document/src/main/java/org/apache/jackrabbit/oak/plugins/document/VersionGCRecommendations.java

```diff
@@ -170,7 +170,7 @@ public class VersionGCRecommendations {
             } else {
                 oldestModifiedDryRunDocTimeStamp.set(fullGCDryRunTimestamp);
             }
-        } else {
+        } else if (fullGCEnabled) {
```

Comment:
We should move this check before the `isFullGCDryRun` check.

Reply:
I was assuming dry run mode wouldn't be enabled if fullgc is disabled - but probably better to protect from that indeed. now done in https://github.com/apache/jackrabbit-oak/pull/1570/commits/0777573384e8d6504ab06eda073d0805a4e8a53a